task_sdk/tests/execution_time/test_supervisor.py

```diff
@@ -764,7 +764,7 @@ def watched_subprocess(self, mocker):
         [
             pytest.param(
                 GetConnection(conn_id="test_conn"),
-                b'{"conn_id":"test_conn","conn_type":"mysql"}\n',
+                b'{"conn_id":"test_conn","conn_type":"mysql","type":"ConnectionResult"}\n',
```

Comment:
We need this for the discriminator to work properly when sending message from Supervisor to Task Runner

Reply:
https://github.com/apache/airflow/blob/09d8a803ca3f76d8598e7eeeb86397cf1cd4076d/task_sdk/src/airflow/sdk/execution_time/comms.py#L93-L96

Reply:
We need to do same for XCom & Variable -- but I will handle those in a separate PR

Reply:
Why, what changed here for us having to do that?

Reply:
>Why, what changed here for us having to do that?

https://github.com/apache/airflow/pull/45043#discussion_r1890524124

It was wrong from the start :) -- We weren't using or sending messages from Supervisor to the Task Runner. We were only checking if Supervisor can handle "that request".

Reply:
Maybe I am getting it wrong, but isn't the purpose of this test to just check if a certain kind of "message" can be handled by the supervisor? supervisor -> client communication essentially. We never touch any part of task runner iiuc.

Reply:
>isn't the purpose of this test to just check if a certain kind of "message" can be handled by the supervisor

That's correct, but we are verifying two things:

1) Supervisor can handle that message i.e. it can consume that message, deserialize it and calls the correct "client" method
2) Supervisor serializes the response to json and we assert the expected serialized json message that should be added to stdin

Those are also in docstrings:

https://github.com/apache/airflow/blob/3c111680a837f22659555c9b6647df40d6ead5f0/task_sdk/tests/execution_time/test_supervisor.py#L901-L905

Reply:
Gotcha, thanks